akka-stream-tests/src/test/scala/akka/stream/remote/StreamRefsSpec.scala

```diff
@@ -0,0 +1,207 @@
+      //        val ref: SourceRef[String] = source.runWith(SourceRef())
+      //
+      //        sender() ! SourceMsg(ref)
+      //
+      //      case "send-bulk" ⇒
+      //        /*
+      //         * Here we're able to send a source to a remote recipient
+      //         * The source is a "bulk transfer one, in which we're ready to send a lot of data"
+      //         *
+      //         * For them it's a Source; for us it is a Sink we run data "into"
+      //         */
+      //        val source: Source[ByteString, NotUsed] = Source.single(ByteString("huge-file-"))
+      //        val ref: SourceRef[ByteString] = source.runWith(SourceRef.bulkTransfer())
+      //        sender() ! BulkSourceMsg(ref)
+
+      case "receive" ⇒
+        /*
+         * We write out code, knowing that the other side will stream the data into it.
+         *
+         * For them it's a Sink; for us it's a Source.
+         */
+        val sink: Future[SinkRef[String]] =
+          SinkRef.source[String]
+            .to(Sink.actorRef(probe, "<COMPLETE>"))
+            .run()
+
+        // FIXME we want to avoid forcing people to do the Future here
+        sender() ! Await.result(sink, 10.seconds)
+
+      //      case "receive-bulk" ⇒
+      //        /*
+      //         * We write out code, knowing that the other side will stream the data into it.
+      //         * This will open a dedicated connection per transfer.
+      //         *
+      //         * For them it's a Sink; for us it's a Source.
+      //         */
+      //        val sink: SinkRef[ByteString] =
+      //          SinkRef.bulkTransferSource()
+      //            .to(Sink.actorRef(probe, "<COMPLETE>"))
+      //            .run()
+      //
+      //
+      //        sender() ! BulkSinkMsg(sink)
+    }
+
+  }
+
+  // -------------------------
+
+  final case class SourceMsg(dataSource: SourceRef[String])
+  final case class BulkSourceMsg(dataSource: SourceRef[ByteString])
+  final case class SinkMsg(dataSink: SinkRef[String])
+  final case class BulkSinkMsg(dataSink: SinkRef[ByteString])
+
+  def config(): Config = {
+    val address = SocketUtil.temporaryServerAddress()
+    ConfigFactory.parseString(
+      s"""
+    akka {
+      loglevel = debug
+
+      actor {
+        provider = remote
+        serialize-messages = on
+
+//        serializers {
+//          akka-stream-ref-test = "akka.stream.remote.StreamRefsSpecSerializer"
+//        }
+//
+//        serialization-bindings {
+//          "akka.stream.remote.StreamRefsSpec$$SourceMsg" = akka-stream-ref-test
+//          "akka.stream.remote.StreamRefsSpec$$BulkSourceMsg" = akka-stream-ref-test
+//          "akka.stream.remote.StreamRefsSpec$$SinkMsg" = akka-stream-ref-test
+//          "akka.stream.remote.StreamRefsSpec$$BulkSinkMsg" = akka-stream-ref-test
+//        }
+//
+//        serialization-identifiers {
+//          "akka.stream.remote.StreamRefsSpecSerializer" = 33
+//        }
```

Comment:
Not using them currently, sending back a raw SinkRef (which one should not do in reality, but testing the serialization things now)